safeweb/http.go

```diff
@@ -376,10 +337,15 @@ func (s *Server) serveBrowser(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Security-Policy", s.csp)
 	w.Header().Set("X-Content-Type-Options", "nosniff")
 	w.Header().Set("Referer-Policy", "same-origin")
-	if s.SecureContext {
+	if s.ServeHSTS {
 		w.Header().Set("Strict-Transport-Security", cmp.Or(s.StrictTransportSecurityOptions, DefaultStrictTransportSecurityOptions))
 	}
-	s.csrfProtect(s.BrowserMux).ServeHTTP(w, r)
+	if !slices.Contains([]string{"GET", "HEAD", "OPTIONS"}, r.Method) && r.Header.Get("Sec-Fetch-Site") != "same-origin" {
+		// deny non-GET/HEAD/OPTIONS requests that are not same-origin
+		http.Error(w, "forbidden cross-origin request", http.StatusForbidden)
```

Comment:
let's return a different error if `Sec-Fetch-Site` is not sent by the client instructing them to update their browser